EIPS/eip-6900.md

```diff
@@ -0,0 +1,358 @@
+        bytes4 implSelector;
+    }
+
+    event PluginsUpdated(PluginUpdate[] pluginUpdates, address init, bytes callData);
+
+    event GlobalPluginsUpdated(AssociatedFunction[] globalPluginUpdates, address init, bytes callData);
+
+    /**
+        * @notice Add/replace/remove any number of plugins and optionally execute a function
+        * @param pluginUpdates Contains the plugin addresses and function selectors.
+        *        executionPluginAddress specifies the plugin containing the execution functions defined within
+        *        pluginAction denotes what operation to perform
+        *        executionUpdates denote which execution function and associated function to perform the opeartion on.
+        * @param init The address of the contract or facet to execute calldata
+        * @param callData A function call, including function selector and arguments
+        */
+    function updatePlugins(PluginUpdate[] memory pluginUpdates, address init, bytes calldata callData) external;
+
+    /**
+     * @notice Add/replace/remove any number of global plugins and optionally execute a function
+     * @param globalPluginUpdates Contains the plugin addresses and function selectors.
+     * @param init The address of the contract or facet to execute calldata
+     * @param callData A function call, including function selector and arguments
+     */
+   function updateGlobalPlugins(GlobalPluginUpdate[] memory globalPluginUpdates, address init, bytes calldata callData)
+       external;
+}
+```
+
+Plugin inspection interface. Modular Smart Contract Accounts MAY implement this interface to support visibility in plugin configuration on-chain. Note this will require extra storage.
+
+#### `IPluginLoupe.sol`
+
+```solidity
+interface IPluginLoupe {
+    struct PluginInfo {
+        address executionPluginAddress;
+        FunctionConfig[] configs;
+    }
+
+    struct FunctionConfig {
+        bytes4 executionSelector;
+        FunctionReference userOpValidator;
+        FunctionReference runtimeValidator;
+        FunctionReference preHook;
+        FunctionReference postHook;
+    }
+
+    struct FunctionReference {
+        address implAddress;
+        bytes4 implSelector;
+    }
+
+    function getPlugins() external view returns (PluginInfo[] memory);
+
+    function getPlugin(address executionPluginAddress) external view returns (FunctionConfig[] memory functions);
+
+    function getFunctionConfig(bytes4 executionSelector) external view returns (FunctionConfig memory);
+}
+```
+
+Plugin interface. Plugins MUST implement the following interface.
+
+#### `IPlugin.sol`
+
+```solidity
+interface IPlugin {
+  // Returns the starting storage slot owned by the plugin.
+  // Storage may extend past this slot, or to other locations
+  // when used through mappings or dynamically-sized arrays
+  function pluginStorageRoot() external view returns (bytes32 slot);
+}
+```
+
+### Expected behavior
+
+**Calls to** `updatePlugins`
```

Comment:
Should these bolded bits be subsection headers?

Something like:

```suggestion
#### **Calls to** `updatePlugins`
```

Reply:
Yes, when originally writing we did not know that this flavor of markdown supports heading levels beyond 3. Just updated.